### PR TITLE
RHEL based Java S2I builder

### PR DESCRIPTION
**Pull request by sabre1041**
#### What does this PR do?
Adds RHEL based Java S2I builder

#### How should this be manually tested?

Login to an OpenShift environment and create a new or use an existing project

Build the S2I image

`oc new-build https://github.com/sabre1041/containers-quickstarts#s2i-java --docker-image=registry.access.redhat.com/rhel:7.2 --context-dir=s2i-java --name=s2i-java --strategy=docker`

Once the build completes successfully, deploy a sample application

`oc new-app s2i-java~https://github.com/christian-posta/springboot-f8-demo --name=s2i-java-app`

Verify the build completes successfully and the application is deployed.

Expose the *s2i-java-app* so that is externally accessible

`oc expose svc s2i-java-app`

Locate the hostname of the route that was created

`oc get route s2i-java-app --template='{{ .spec.host }}'`

Navigate to the following URL to confirm functionality

`http://<hostname>/api/hello/test`

A JSON response similar to the following should be returned

```
{"response":"hello","count":0,"your-name":"test"}
```

#### Is there a relevant Issue open for this?
https://trello.com/c/dXlPwQWI

#### Who would you like to review this?
/cc @etsauer  @raffaelespazzoli

### CODE REVIEW

**Comment by etsauer** [comment]
@sabre1041 oh, one other thing. Can we include an app template?

**Comment by raffaelespazzoli** [comment]
I'm good with this pr.
One improvement that we have discussed on slack and is valid for both s2i-java and s2i-liberty is not to expose debug and jolokia ports with the services. There doesn't seem to be a way to do it, maybe an opportunity for an RFE.

File: s2i-java/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+ENV MAVEN_VERSION="3.3.3" \
+    JOLOKIA_VERSION="1.3.5" \
+    PATH=$PATH:"/usr/local/s2i" \
+    AB_JOLOKIA_CONFIG="/opt/jolokia/jolokia.properties" \
+    AB_JOLOKIA_AUTH_OPENSHIFT="true" \
+    JAVA_DATA_DIR=/deployments/data
+
+# Some version information
+LABEL io.k8s.description="Platform for building and running plain Java applications (fat-jar and flat classpath)" \
```
Comment (sabre1041) [inline s2i-java/Dockerfile]: I think we should have a cautionary message. but lets discuss how we want to implement this with the team

**Comment by sabre1041** [comment]
@etsauer I have added a RHEL template. Subsequent work can go in to develop a centos based image and template

Also, the combination of the template and the inclusion of the `jolokia` container port enables the Java web console from the pods page of the UI which enables access to the jolokia web console

**Comment by etsauer** [comment]
lgtm. merging.

**Pull request merged**